server/service/src/sync/api/error.rs

```diff
@@ -1,58 +1,128 @@
 use super::*;
```

Comment:
In this file change SyncApiError to have url, then added SyncApiErroVariant (as previous SyncApiError), and flattened mapped error into SyncApiVariant.
Also added concrete error code mapping form central server